plugins/specialchar/dialogs/specialchar.js

```diff
@@ -29,7 +29,7 @@ CKEDITOR.dialog.add( 'specialchar', function( editor ) {
 
 			// We must use "insertText" here to keep text styled.
 			var span = editor.document.createElement( 'span' );
-			span.setHtml( value );
+			span.setHtml( CKEDITOR.tools.htmlSafeByReview(CKEDITOR.tools.htmlEncode(value), 'Value is the html of an element that already exists on the page that has triggered some event. We know event data is not very safe for these purposes and it looks like this is pulling the element html directly so we may want to legacy convert.') );
```

Comment:
For this instance it looks like span is only used temporarily and the text is pulled out of it using span.getText() which uses innerText or textContent so it would likely be safe.

Reply:
Sorry, to clarify, this `span` that is created here is not attached to the document? Does it have XSS risk?

Regardless, I see that it's passing through a `htmlEncode` here-- how much does that "sanitize" the HTML, and does it neutralize enough of the tags that we can consider just passing it through another round of sanitization without breaking the content that's already in `value`?

Reply:
I guess if we're taking away angle-brackets then it should be fine!